### PR TITLE
BOM-2948: Replace Travis with GitHub CI

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**Issue:** [BOM-2948](https://openedx.atlassian.net/browse/BOM-2948)

### Description
- Replaced `Travis` with `GitHub CI`.
- Coverage check was already not working on `Travis` because this repo hasn't been setup on `codecov.io` yet so skipped the coverage check in `GitHub CI`.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - name: setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: pip install -r requirements/ci.txt
+
+      - name: Run Tests
+        run: tox
```
Comment (estute) [inline .github/workflows/ci.yml:32]: In the .travis.yml file, we also ran codecov after the tests. Do we still want to do that?

**Review by estute — COMMENTED** [review]
See inline

**Review by estute — APPROVED** [review]
One question, but otherwise looks good to me.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kdmccormick) [reply .github/workflows/ci.yml:32]: Added

**Comment by kdmccormick** [comment]
@estute I went and added the codecov check in a separate commit, but it fails currently.

Looking at [previous builds](https://app.travis-ci.com/github/edx/user-util/builds/241759731), it looks like the `codecov` invocation just silently fails with `codecov: command not found`. 😦 The Python codecov package was never installed in Travis CI requirements.

I am going to revert my codecov commit and just merge this PR as it was in order to unblock the de-Travisification effort, but heads up that your team may want to install and configure codecov if that's something that you folks see as valuable in this repo.

**Comment by estute** [comment]
@kdmccormick thanks for tracking that down.

**Pull request merged**